### PR TITLE
use yari's *tool* as *content* instead of redirects for validation

### PR DESCRIPTION
**Pull request by fiji-flo**
Missing the updated yari (need to be published first).

### CODE REVIEW

**Comment by peterbe** [comment]
I'm not convinced that it's named as well as it could be. If only it was `yarn redirect-validate ...` instead of `yarn tool redirect-validate ...`
It's just an odd word and somehow the grammar feels weird. With "yarn redirect-validate" it reads as "NOUN VERB".
Now, perhaps I'm just hung up on the word "tool" because it's technically a namespace here. And having it like that makes it convenient for us because we don't need to write down every single script (after the `yarn build ...` thing). 
Perhaps if it was `yarn content validate-redirect` it would feel more "natural". It's related to the *content*, as opposed to the test server or git workflow for example. 

Perhaps it's too late now but, it reminds me that "validate-redirect" reads more fluidly and naturally than "redirect-validate". 

I think the code is almost unimportant. What matters is how these commands would *feel* in the hands of editors and content writers who bother git cloning mdn/content. 
So as of that, I think we should collect impressions and gut-preferences from @chrisdavidmills and @escattone and @ddbeck

**Comment by escattone** [comment]
@peterbe I agree. I think in this repo the commands should be intuitive as possible for content contributors.

@peterbe @fiji-flo @schalkneethling @ddbeck @chrisdavidmills What do you all think if we used the following convenience (syntactical sugar) commands?

```json
"scripts": {
  ...
  "validate-redirect": "env-cmd --silent cross-env CONTENT_ROOT=files node node_modules/@mdn/yari/tool/cli.js redirect-validate",
  "test-redirect": "env-cmd --silent cross-env CONTENT_ROOT=files node node_modules/@mdn/yari/tool/cli.js redirect-test",
  "add-redirect": "env-cmd --silent cross-env CONTENT_ROOT=files node node_modules/@mdn/yari/tool/cli.js redirect-add",
  "delete": "env-cmd --silent cross-env CONTENT_ROOT=files node node_modules/@mdn/yari/tool/cli.js content-delete",
  "move": "env-cmd --silent cross-env CONTENT_ROOT=files node node_modules/@mdn/yari/tool/cli.js content-move",
  "edit": "env-cmd --silent cross-env CONTENT_ROOT=files node node_modules/@mdn/yari/tool/cli.js content-edit",
  "create": "env-cmd --silent cross-env CONTENT_ROOT=files node node_modules/@mdn/yari/tool/cli.js content-create",
  "validate": "env-cmd --silent cross-env CONTENT_ROOT=files node node_modules/@mdn/yari/tool/cli.js content-validate",
  "preview": "env-cmd --silent cross-env CONTENT_ROOT=files node node_modules/@mdn/yari/tool/cli.js content-preview"
  },
```

**Comment by fiji-flo** [comment]
We can also rename the command in the tool directly. It keeps things simpler if we just expose the whole tool here. But again as with all the names, no strong opinion.

After my painful migration of _init.d_ (`service ssh start`) to systemd (`systemctl start sshd`) I lost all opinions :)

**Comment by escattone** [comment]
> After my painful migration of _init.d_ (`service ssh start`) to systemd (`systemctl start sshd`) I lost all opinions :)
😆

**Comment by escattone** [comment]
@fiji-flo @peterbe @ddbeck @schalkneethling @chrisdavidmills 
> We can also rename the command in the tool directly. It keeps things simpler if we just expose the whole tool here. But again as with all the names, no strong opinion.

I like that idea too. If we did that, what do you all think about changing the name of `tool` to `content` at https://github.com/mdn/yari/blob/378713a134dda3059ae219318ac0d4c2e92e128e/package.json#L27, and then changing the commands to:
- `redirect-validate` --> `validate-redirect`
- `redirect-test` --> `test-redirect`
- `redirect-add` --> `add-redirect`
- `content-delete` --> `delete`
- `content-move` --> `move`
- `content-edit` --> `edit`
- `content-create` --> `create`
- `content-validate` --> `validate`
- `content-preview` --> `preview`

This may have been what @peterbe has been proposing all along?

**Comment by chrisdavidmills** [comment]
I like the structure and naming consistency of these commands. I don't see anything else that I really think should be there by the beta test. I'm assuming the `content-validate` command runs some kind of linting machinery on the changes?

**Comment by peterbe** [comment]
> I like the structure and naming consistency of these commands. I don't see anything else that I really think should be there by the beta test. I'm assuming the `content-validate` command runs some kind of linting machinery on the changes?

No it does not run the kind of linting you're probably thinking of. For that, you have to *build* the doc. 
This thing just checks that a given a `slug` is what's case-insensitively is what's in the front-matter. 
```
▶ node tool/cli.js content-validate mdn/kitChensink
✓ All seems fine
```
https://github.com/mdn/yari/blob/29282cbe866139b4e755dbc9fa8b6dbac35c579d/content/document.js#L376-L390

I actually don't know what use this is to writers. Perhaps @fiji-flo can explain or even for us. 

@fiji-flo Would you have time to jot down a couple of notes, yari, in  `docs/cli-tool.md` file? It's where we maintain some docs for ourselves and our future devs of the platform.

**Review by escattone — APPROVED** [review]
@fiji-flo Thumbs up from me!

File: .github/workflows/pr-build.yml [hunk 1]
```diff
@@ -147,4 +147,4 @@ jobs:
         run: |
           echo ${{ env.GIT_DIFF }}
           echo ${{ env.MATCHED_FILES }}
-          yarn redirects validate
+          yarn content validate-redirects
```
Comment (peterbe) [inline .github/workflows/pr-build.yml:150]: Beautiful!

**Review by peterbe — APPROVED** [review]

**Pull request merged**